doc: added release notes for v1.8 draft new

Shilpashree3lc-zz opened the pull request:
Draft of Zephyr 1.8 release notes which contains some interesting
changes on Kernel, Architecture, Boards, Drivers and Sensors,
Networking, Bluetooth, documentation, and jira items.etc.

Signed-off-by: LC Shilpashree shilpashree.lc@intel.com

Commit:
dts/extract_dts_includes.py: Add partition support

This patch adds parsing support for flash partitions.

Signed-off-by: Andy Gross <andy.gross@linaro.org>

Commit:
kconfig: misc: Conditionally remove FLASH_LOAD options

This patch adds a condition on the FLASH_LOAD_OFFSET and FLASH_LOAD_SIZE
to make them conditionally excluded if HAS_DTS is enabled.  FLASH
partition information must be defined for all DTS platforms which
utilize flash.

For DTS enabled platforms, CONFIG_FLASH_LOAD_OFFSET and
CONFIG_FLASH_LOAD_SIZE will be generated from the flash information
defined in the DTS file.  The values used for these variables will be
determined by the zephyr,code-partition chosen node.  If no chosen
node is specified, the zephyr,flash chosen node will be used.

Signed-off-by: Andy Gross <andy.gross@linaro.org>

Commit:
dts: common: yaml: Add partition file

This patch adds the yaml definition for flash partitions.

Signed-off-by: Andy Gross <andy.gross@linaro.org>

Commit:
frdm_k64f: default to pyocd.sh for flashing/debug

The wiki directions indicate that this script should be used,
and openocd.sh doesn't even work. Switch to pyocd.sh by default.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
stack_sentinel: change cooperative check

One of the stack sentinel policies was to check the sentinel
any time a cooperative context switch is done (i.e, _Swap is
called).

This was done by adding a hook to _check_stack_sentinel in
every arch's __swap function.

This way is cleaner as we just have the hook in one inline
function rather than implemented in several different assembly
dialects.

The check upon interrupt is now made unconditionally rather
than checking if we are calling __swap, since the check now
is only called on cooperative _Swap(). The interrupt is always
serviced first.

Issue: ZEP-2244
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
arches: declare _SysFatalErrorHandler __weak

This function is intended to be easily overridable by applications.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
tests: fatal: increase coverage

- _SysFatalErrorHandler is supposed to be user-overridable.
The test case now installs its own handler to show that this
has happened properly.

- Use TC_PRINT() TC_ERROR() macros

- Since we have out own _SysFatalErrorHandler, show that
k_panic() works

- Show that _SysFatalErrorHandler gets invoked with the expected
reason code for some of the scenarios.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
k_oops: force unlock IRQs on ARMv7M

Fixes an issue where if a thread calls k_panic() or k_oops()
with interrupts locked, control would return to the thread
and it would only be aborted after interrupts were unlocked
again.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
stack_sentinel: hang system on failure

Stack sentinel doesn't prevent corruption, it just notices when
it happens. Any memory could be in a bad state and it's more
appropriate to take the entire system down rather than just kill
the thread.

Fatal testcase will still work since it installs its own
_SysFatalErrorHandler.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
samples: static_lib: conditional assign BOARD (?=)

The sample app "static_lib" is very important to Zephyr user, which
demonstrate how to build and link a static lib.
ISSM team wanted to integrate this app in their IDE for quark platforms.
However they find the in "static_lib/hello_world/Makefile" BOARD is
hardcoded as qemu_x86.

This patch supports other BOARD passed from build command.
I have verified this app working fine @Arduino101.

Signed-off-by: Sharron LIU <sharron.liu@intel.com>

nashif committed:
license: add missing licenses and copyright

We were missing license boilerplate in many files, add them

Jira: ZEP-1464

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: remove obsolete usage of defrag

Also increase ISR stack to make it run on Quark D2000 CRB.

Jira: ZEP-2224
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

galak committed:
toolchain.gccarmemb: Fix support for where to find newlib

When we build with newlib enabled and utilizing one of the other
variants (like having floating point enabled) we need to have the proper
library path setup to find the library.  This is mimicked after what we
do in Makefile.toolchain.zephyr for newlib.

Issue: ZEP-2240

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

Commit:
sam3x: report correct number of IRQ priority bits

The Sam3x HAL defines __NVIC_PRIO_BITS to 4.
Fixes an issue where interrupt priorities and masking
were not being done correctly.

Issue: ZEP-2243
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
tests: context: move idle test to the end

On some devices, when k_cpu_idle() was called we were getting
interrupts that were not the timer interrupt. On bbc_micro
a power clock control driver interrupt was happening instead
and k_cpu_idle() was returning without the system tick advancing,
failing the test.

The clock control interrupts seem to only happen early in device
boot; moving the idle test much later lets the test pass on this
board (and likely all other NRF5 based boards).

Issue: ZEP-2257
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

dbkinder committed:
doc: starting draft for 1.8 release notes

Draft of 1.8 release notes with heading and jira items from
draft doc. Added 1.8 release notes to index.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

dbkinder commented:
This PR should only have one file being committed (doc/release-notes-1.8.rst)?

dbkinder commented on doc/release-notes-1.8.rst:
```diff
@@ -95,20 +190,27 @@ JIRA Related Items
 * :jira:`ZEP-932` - Adapt kernel sample & test projects
 * :jira:`ZEP-948` - Revisit the timeslicing algorithm
-* :jira:`ZEP-973` - Remove deprecated API related to device PM, DEVICE\_ and * SYS\_* macros
+* :jira:`ZEP-973` - Remove deprecated API related to device PM, DEVICE_ and * SYS_* macros
```
DEVICE\_ and * SYS\_*  (need to escape the _ to keep Sphinx from thinking these are label reference links)

nashif reviewed (changes requested):
Please the PR and make sure you are changing the last version of the release notes available in the branch, we received a few updates already.

nashif commented:
superseded by #491

Shilpashreelc3 committed:
resolve conflict by keeping changes done by someone else

Shilpashreelc3 committed:
doc: added release notes for v1.8 draft new

Draft of Zephyr 1.8 release notes which contains some interesting
changes on Kernel, Architecture, Boards, Drivers and Sensors,
Networking, Bluetooth, documentation, and jira items.etc.

Signed-off-by: LC Shilpashree shilpashree.lc@intel.com
